### PR TITLE
fix(auth): hand off qr bootstrap to bounded device tokens

### PR DESCRIPTION
**Pull request by ngutman**
## Summary

- Problem: iOS QR bootstrap onboarding could connect the node session, but it did not hand off the bounded operator auth needed for post-onboarding operator-capable flows.
- Why it matters: fresh QR-only onboarding could not immediately participate in approval-gated operator flows, and visible push permission was easy to miss after a reinstall.
- What changed: QR bootstrap now seeds bounded `node` + `operator` device tokens, persists only the trusted bootstrap handoff tokens on iOS, and requests notification authorization immediately after successful QR onboarding.
- What did NOT change (scope boundary): no exec approval push/modal UI in this PR; no approval inbox/history UI; no widening to `operator.admin` or `operator.pairing`.

## Change Type (select all)

- [x] Bug fix
- [ ] Feature
- [x] Refactor required for the fix
- [ ] Docs
- [x] Security hardening
- [ ] Chore/infra

## Scope (select all touched areas)

- [x] Gateway / orchestration
- [ ] Skills / tool execution
- [x] Auth / tokens
- [ ] Memory / storage
- [ ] Integrations
- [x] API / contracts
- [x] UI / DX
- [ ] CI/CD / infra

## Linked Issue/PR

- Closes #
- Related #59732
- [x] This PR fixes a bug or regression

## Root Cause / Regression History (if applicable)

- Root cause: QR setup code/bootstrap auth was intentionally node-bootstrap-oriented and iOS refused to start the operator loop from bootstrap-only auth.
- Missing detection / guardrail: we did not return/store bounded per-role device tokens from the successful bootstrap handshake.
- Prior context (`git blame`, prior PR, issue, or refactor if known): split out from the broader iOS exec approval work previously carried on `ios/exec-approval-push` / PR #59732.
- Why this regressed now: N/A; this is a missing onboarding handoff rather than a newly introduced regression on a shipped iOS user base.
- If unknown, what was ruled out: generic APNs transport and QR pairing itself were ruled out during manual device testing.

## Regression Test Plan (if applicable)

- Coverage level that should have caught this:
  - [x] Unit test
  - [x] Seam / integration test
  - [ ] End-to-end test
  - [ ] Existing coverage already sufficient
- Target test or file:
  - `src/infra/device-bootstrap.test.ts`
  - `src/infra/device-pairing.test.ts`
  - `src/gateway/server.auth.control-ui.test.ts`
  - `src/pairing/setup-code.test.ts`
  - `apps/shared/OpenClawKit/Tests/OpenClawKitTests/GatewayNodeSessionTests.swift`
  - `apps/ios/Tests/GatewayConnectionControllerTests.swift`
  - `apps/ios/Tests/NodeAppModelInvokeTests.swift`
- Scenario the test should lock in: QR bootstrap can only seed the bounded baseline, returns/stores `node` + `operator` device tokens, and requests notification authorization after successful bootstrap onboarding.
- Why this is the smallest reliable guardrail: the issue crosses gateway auth, device pairing persistence, and iOS onboarding state.
- Existing test that already covers this (if any): none covered the full handoff before this change.
- If no new test is added, why not: N/A.

## User-visible / Behavior Changes

- Fresh QR onboarding on iOS now silently hands off bounded `node` + `operator` device auth.
- The stored operator token includes `operator.approvals`.
- iOS requests notification permission immediately after successful QR onboarding instead of waiting for a later local-notification path.

## Diagram (if applicable)

```text
Before:
[scan QR] -> [bootstrap node connect only] -> [no stored operator auth] -> [operator flows still blocked]

After:
[scan QR] -> [bootstrap node connect] -> [gateway seeds bounded node+operator tokens] -> [iOS stores both] -> [operator loop can start] -> [notification auth prompt shown]
```

## Security Impact (required)

- New permissions/capabilities? (`Yes/No`) Yes
- Secrets/tokens handling changed? (`Yes/No`) Yes
- New/changed network calls? (`Yes/No`) No
- Command/tool execution surface changed? (`Yes/No`) No
- Data access scope changed? (`Yes/No`) Yes
- If any `Yes`, explain risk + mitigation:
  - Risk: QR bootstrap now seeds operator-capable auth in addition to node auth.
  - Mitigation: the seeded baseline is explicitly bounded to `operator.approvals`, `operator.read`, `operator.talk.secrets`, and `operator.write`; bootstrap remains single-use; public-key binding is normalized; iOS only persists handoff tokens for trusted bootstrap contexts (`wss` or loopback) and filters roles/scopes before storing.

### Environment

- OS: macOS 26.3.1
- Runtime/container: Node 24.x, local gateway
- Model/provider: N/A
- Integration/channel (if any): iOS app / gateway QR onboarding
- Relevant config (redacted): isolated local gateway with QR bootstrap setup

### Steps

1. Generate a fresh QR setup payload and bootstrap token.
2. Connect the iOS app via QR bootstrap.
3. Inspect paired device roles/tokens and confirm operator handoff and notification auth prompt behavior.

### Expected

- The paired iOS device gets bounded `node` + `operator` auth, with `operator.approvals` present.
- The bootstrap token is single-use.
- Notification authorization is requested immediately after successful onboarding.

### Actual

- Matches expected with the changes in this PR.

## Evidence

Attach at least one:

- [x] Failing test/log before + passing after
- [x] Trace/log snippets
- [ ] Screenshot/recording
- [ ] Perf numbers (if relevant)

## Human Verification (required)

- Verified scenarios:
  - reviewed the branch scope against the intended QR-bootstrap-only split
  - validated targeted bootstrap/device-token tests
  - validated targeted iOS simulator tests for operator scope baseline and post-bootstrap notification auth prompt
- Edge cases checked:
  - bootstrap token single-use / binding
  - bounded token persistence filtering
  - non-operator token scope normalization
- What you did **not** verify:
  - a new real-phone rerun on this split branch alone

## Review Conversations

- [x] I replied to or resolved every bot review conversation I addressed in this PR.
- [x] I left unresolved only the conversations that still need reviewer or maintainer judgment.

## Compatibility / Migration

- Backward compatible? (`Yes/No`) Yes
- Config/env changes? (`Yes/No`) No
- Migration needed? (`Yes/No`) No
- If yes, exact upgrade steps:

## Risks and Mitigations

- Risk: storing extra bootstrap handoff tokens too broadly on iOS.
  - Mitigation: only persist on trusted bootstrap transports and aggressively filter the allowed roles/scopes.
- Risk: widening QR bootstrap beyond the minimum needed baseline.
  - Mitigation: keep the bootstrap profile narrowly bounded and do not add `operator.admin` or `operator.pairing`.

### CODE REVIEW

**Comment by ngutman** [comment]
Addressed the review feedback in 1e3af8ad3e.

What changed:
- preserved backward-compatible operator reconnect scopes on iOS by only requesting `operator.approvals` when the current auth path can already support it (shared token/password or a stored operator token that already carries that scope)
- removed the unused/confusing `approvedScopesOverride` path from `src/infra/device-pairing.ts`
- documented the `getBoundDeviceBootstrapProfile(...)` precondition
- stopped duplicating the primary node token in `hello-ok.auth.deviceTokens`
- added regression coverage for the iOS scope-selection behavior and for the bootstrap hello payload expectations

Validation rerun:
- `pnpm test -- src/infra/device-bootstrap.test.ts src/infra/device-pairing.test.ts src/gateway/server.auth.control-ui.test.ts -t 'auto-approves fresh node bootstrap pairing from qr setup code|fails closed for operator approvals when caller scopes are omitted|allows operator scope subsets within the issued bootstrap profile|stores and reads explicit bootstrap profiles'`
- `xcodegen generate` in `apps/ios`
- `xcodebuild -project apps/ios/OpenClaw.xcodeproj -scheme OpenClaw -destination 'platform=iOS Simulator,name=iPhone 16,OS=18.6' -only-testing:'OpenClawTests/GatewayConnectionControllerTests/operatorConnectOptionsOnlyRequestApprovalScopeWhenEnabled()' -only-testing:'OpenClawTests/GatewayConnectionControllerTests/operatorApprovalScopeRequestsStayBackwardCompatible()' -only-testing:'OpenClawTests/NodeAppModelInvokeTests/operatorLoopWaitsForBootstrapHandoffBeforeUsingStoredToken()' test`
- `pnpm build`
- `pnpm check`

Note: `swift test --package-path apps/shared/OpenClawKit --filter GatewayNodeSessionTests/bootstrapHelloStoresAdditionalDeviceTokens` currently runs into unrelated preexisting `TalkSystemSpeechSynthesizerTests` actor-isolation compile errors in the package test target, so I did not treat that as caused by this PR.

@greptile-apps and @chatgpt-codex-connector please re-review this PR.

File: apps/ios/Sources/Model/NodeAppModel.swift [hunk 1]
```diff
@@ -2246,10 +2269,47 @@ private extension NodeAppModel {
...
+
+    nonisolated static func shouldRequestOperatorApprovalScope(
+        token: String?,
+        password: String?,
+        storedOperatorScopes: [String]
+    ) -> Bool {
+        let trimmedToken = token?.trimmingCharacters(in: .whitespacesAndNewlines) ?? ""
+        if !trimmedToken.isEmpty {
+            return true
+        }
```
[the remark this answers is not in the document]
Comment (ngutman) [reply apps/ios/Sources/Model/NodeAppModel.swift:2291]: Thanks — this backward-compatibility concern is valid in principle.

For this rollout, though, we do not have an existing installed/paired iOS user base to preserve here, so there are no real legacy iOS operator pairings that were approved without `operator.approvals`.

The intended path for this PR is fresh onboarding, and on that path QR/bootstrap now seeds bounded stored operator auth including `operator.approvals`, so reconnects continue through the stored-token flow.

Given that, I'm intentionally keeping the simpler behavior here instead of adding extra reconnect gating for old shared-token/password iOS installs that do not exist in practice. If we later need to support migration from pre-`operator.approvals` iOS pairings, we can narrow this gate as follow-up.
